destination.go

```diff
@@ -53,10 +56,24 @@ func (d *Destination) Open(ctx context.Context) error {
 }
 
 func (d *Destination) Write(ctx context.Context, record sdk.Record) error {
+	return d.writeInternal(ctx, record, true)
+}
+
+func (d *Destination) writeInternal(ctx context.Context, record sdk.Record, retry bool) error {
 	err := d.Client.Send(
 		record.Key.Bytes(),
 		record.Payload.Bytes(),
 	)
+	// this can happen when the topic doesn't exist and the broker has auto-create enabled
+	// we give it some time to process topic metadata and retry
+	if retry && errors.Is(err, kafka.LeaderNotAvailable) {
+		sdk.Logger(ctx).
+			Info().
+			Err(err).
+			Msg("leader for topic unavailable, will retry")
```

Comment:
Nit: would be nice to add the retry time to the log, particularly since it's 15 seconds long

Reply:
Alternatively, we could retry once every second for _up to_ 15 seconds, to help minimize the amount of time waiting. (Or even use a backoff of sorts, but that may be overkill for this fix 😄 )